### PR TITLE
`deck`: move readiness serve to where we actually start the server

### PR DESCRIPTION
**Pull request by smg247**
This is a follow up to the failed attempt from https://github.com/kubernetes-sigs/prow/pull/308. It makes it more clear that the CSRF logic wraps the server, and starts the readiness endpoint immediately before starting the server.

### CODE REVIEW

File: cmd/deck/main.go [hunk 1]
```diff
@@ -500,13 +497,15 @@ func main() {
 		return
 	}
 
+	var server *http.Server
 	if csrfToken != nil {
 		CSRF := csrf.Protect(csrfToken, csrf.Path("/"), csrf.Secure(!o.allowInsecure))
-		logrus.WithError(http.ListenAndServe(":8080", CSRF(traceHandler(mux)))).Fatal("ListenAndServe returned.")
```
Comment (petr-muller) [inline cmd/deck/main.go:505]: lol sneaky

**Review by petr-muller — APPROVED** [review]

**Comment by petr-muller** [comment]
/hold

Can we remove the `--allow-insecure` from the tests again? Or at least from one, to integeration-test both paths?

**Comment by smg247** [comment]
> Can we remove the `--allow-insecure` from the tests again? Or at least from one, to integeration-test both paths?

Since the other PR was reverted, this has required no changes to the integration tests at all. It is not present in the `deck_tenant_deployment` now.

File: cmd/deck/main.go (hunk 1 above)
Comment (smg247) [reply cmd/deck/main.go:505]: Really, it's like it was written to deceive

**Comment by petr-muller** [comment]
> Since the other PR was reverted, this has required no changes to the integration tests at all. It is not present in the deck_tenant_deployment now.

Ugh, I forgot that because we reverted the test manifests are now in the state I wanted. Code reviews on Friday :grimacing:

**Pull request merged**